### ISSUE TITLE
CBA_fnc_compatibleItems Error

### ISSUE THREAD

**Issue by Mike-MF** [CBATeam/CBA_A3#1611]
**Mods (complete and add to the following information):**
- **Arma 3:** `2.14`
- **CBA:** `3.16.0-RC1`

**Description:**
Attempting to call `fnc_compatibleItems` (in scheduled) returns:

```
if (_typefilter isEqualType "") then {
_type>
 2:59:35   Error position: <_typefilter isEqualType "") then {
_type>
 2:59:35   Error Undefined variable in expression: _typefilter
 2:59:35 File /x/cba/addons/jr/fnc_compatibleItems.sqf..., line 260
 2:59:35 Error in expression <eapon] call BIS_fnc_error;

[]
};
```
It does work just fine unscheduled.

**Steps to reproduce:**
- [CLASSNAME] call CBA_fnc_compatibleItems
- Check RPT.

**Expected behavior:**
Should return the compatible items.

**Where did the issue occur?**
- Dedicated / Self-Hosted Multiplayer / Singleplayer / Editor (Singleplayer) / Editor (Multiplayer) / Virtual Arsenal

**Additional context:**
Error comes from #1558

**Comment by jonpas** [comment]
@johnb432

**Comment by PabstMirror** [comment]
when called from scheduled

**Comment by johnb432** [comment]
See #1612.

**Comment by commy2** [comment]
1558 should've never been approved and should be reverted entirely.

**Comment by jonpas** [comment]
Yah, I think I got it mixed up with some other PR when cleaning things up for the release. That was not properly reviewed.

**Comment by jonpas** [comment]
Fixed by #1612.
